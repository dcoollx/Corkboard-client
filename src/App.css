.App {
}


.App-header {
  background-color: #282c34;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.container{
  display: grid;
  grid-template-columns: auto auto auto auto auto auto;
  row-gap: 2%;
  column-gap: 10px; 
}
.col-full{
  grid-column: span 6;
}
.col-half{
  grid-column: span 3;
}
.col-right{
  grid-column: 4;
}
col-center{
  grid-column: 2/span 4;
}
