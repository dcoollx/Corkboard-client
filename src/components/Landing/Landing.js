import React from 'react';
import {Link} from 'react-router-dom';
//images
import connections_icon from '../../images/connections.png';
import pictures from '../../images/picture.png';
import browser from '../../images/browser.png';

export default function Landing(){
  return(
    <div id="landing" className="container page">
      <div id="landing_header" className="col-full small_container">
        <div  className="col-full small_container">
          <div className="col-1"><h1 >Corkboard</h1>
          <q>The Inter-Organization Message Management System</q>
          </div>

          <div className="landing_sub col-right">
        New here? <Link to="/register">Create An Account</Link><br/>
        Come here often? <Link to="/login">Login</Link>
        </div>
        </div> 
      </div> 
      <div className="small_container col-full  " >
        <h3 className="col-left">Share messsages with your whole Organization or just a single team</h3>
        <img className="col-right" src={connections_icon} alt="connections"/>
      </div>
      <div className="small_container col-full  " >
      <img className="col-left" src={pictures} alt="pictures"/>
        <h3 className="col-right">Share photos and links</h3>
      </div>
      <div className="small_container col-full  " >
        <h3 className="col-left">Craft exciting notifications using HTML5 or markdown</h3>
        <img className="col-right" src={browser} alt="web browser"/>
      </div>
      <footer className="small_container col-full ">
        <p className = "col-1">
          Demo Credentials : testuser, Password1!
        </p>
       <p id="copyright" className="col-right"> &copy; 2019 <a href="https://dcoollx.github.io/Portfolio">David Queen Jr</a></p>
      </footer>
      
    </div>
  );
}