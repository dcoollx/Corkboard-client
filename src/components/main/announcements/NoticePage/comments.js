import React from 'react';


export default function(props){
  return(
  <div>
    <p>"{props.content}"</p>
    <p>{props.by}</p>
  </div>);
}