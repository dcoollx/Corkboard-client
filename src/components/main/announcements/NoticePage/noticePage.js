import React from 'react';
import Comments from './comments';
import {testComments} from '../../../../test/demoData';


export default function(props){
  //todo, make this a fetch to get commets from server, for now using testdata
  let comments = testComments(props.notice.id).map((comment, index)=>{
    return (<Comments key={index} content={comment.content} by={comment.created_by}/>);
  });
  return(
  <div>
    <h2>{props.notice.title}</h2>
    <p>{props.notice.content}</p>
    <p>by: <i>{props.notice.created_by}</i></p>
    {comments}
    <button onClick={(e)=>props.history.push('/')}>Go Back</button>
  </div>);
}