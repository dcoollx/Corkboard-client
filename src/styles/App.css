@import './import.styles.css';
@import url('https://fonts.googleapis.com/css?family=Nunito&display=swap');

main, #root, .App, html, body{
  height: 100vh;
  width: 100vw;
  background-image: url('../images/corkboard_img.jpg');
  font-size: 36px;
  font-family: 'Nunito', sans-serif;
}
input, textarea, select{
  line-height: 36px;
  font-size: 32px;
}

button{
  border: grey 1px solid;
  box-shadow: none;
  border-radius: 20px;
  background-color:var(--accent-light);
  color:white;
  line-height: 80px;
  font-size: 36px;
}
h1, h2, h3, h4{
  
}
.error{
  color: red;
  font-size: 50%;
}
.main-container{
  display: grid;
  grid-template-columns: 20% auto auto auto 20%;
  grid-template-rows: 15vh auto auto auto auto auto auto auto  ;
  row-gap: 5%; 
}
#copyright{
  bottom: 0px;
}
.small_container{
  display:grid;
  grid-template-columns: auto auto auto;
  row-gap: 0px;
}
.container{
  display: grid;
  grid-template-columns: auto auto auto auto auto;
  row-gap: 0px; 
}
.col-full{
  grid-column: span 5;
}
.col-center,.col-1, .col-half, .col-left, .col-right{
  grid-column: 2/span 3;
}
@media only screen and (min-width:640px){

  .main-container{
    display: grid;
    width: 100vw;
    grid-template-columns:  auto auto auto auto auto ;
    grid-template-rows: 25vh auto auto auto auto auto auto auto  ;
    row-gap: 2%;
    column-gap: 10px; 
  }
.col-left{
  grid-column: 1;
  }
.col-full{
  grid-column: span 5;
}
.row-full{
    grid-row-start: 3;
    grid-row-end: auto;
}
.col-half{
  grid-column: span 2;
}
.col-right{
  grid-column: 5;
}
.col-center{
  grid-column: 2/span 3;
}
.col-1{
  grid-column: span 1;
}
.row-center{
  grid-row: 4
}
}
textarea{
  width:100%;
}
body{
--primary:#B28B54;
--secondary:#664B26;
--nuetral:#FFD191;
--accent-dark:#1C4966;
--accent-light:#548EB3;
--shadow:rgba(115, 89, 54, 0.45);
}