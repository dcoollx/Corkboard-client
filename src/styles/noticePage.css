#noticePage{
  background-color: white;
  box-shadow: 5px 5px var(--shadow);
  width:90vw;
  height:100vh;
  overflow: scroll;
  padding: 25px;
  grid-column-start: 1;
  
}
#noticePage-controls{
  width:100%;
}
.comments{
  font-size: 50%;
}


@media only screen and (min-width:640px){
#noticePage{
  width:100%;
  grid-column: 2/span 3;

}
.content{
  padding:30px;

}
}
