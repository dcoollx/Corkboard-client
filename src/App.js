import React from 'react';
import {Route, Switch} from 'react-router-dom';
import './App.css';
import Main from './components/main/main';
import Login from './components/login/login';
import NoticePage from './components/main/announcements/NoticePage/noticePage';
import NewNotice from './components/newNotice/newNotice';
import {testNotices} from './test/demoData';
import token_service from './services/token.service';
import Api from './services/api.service';

class App extends React.Component {
  state = {
    org:{
       orgName:null,
        orgId:null
    },
    notices:[...testNotices(5)],
 
    hasError:false,
    err:null
  };
  
  componentDidMount(){
    let options ={
      method:'GET',
      headers: new Headers({'content-type':'application/json'})
    };
    Api.doFetch('1/corkboard',options)
      .then(notices=>this.setState({notices:notices.notices}))
      .catch(err=>{
        this.setState({hasError:true,err})
        console.log('fetch encountered an error', err);
      });
  }
  signIn = (user_name, password,org)=>{
    let options ={
      method:'POST',
      headers:new Headers({'Content-type':'application/json'}),
      body: JSON.stringify({user_name,password,org})
    };
    return Api.doFetch('login',options).then((res)=>{
      token_service.setAuthToken(res.Auth);
      this.setState({org:{orgName:org,orgId:1}})//TODO, orgid is hardcoded
      localStorage.setItem('orgInfo',{orgName:org,orgId:1})
    });
  }
  render(){
    console.log(this.state.notices)
  return (
    <div className="App">
    <Switch>
      <Route exact path="/" render={(props)=>{
        if(token_service.hasAuthToken())
          return(
          <Main {...props} orgInfo = {this.state.org} announcements={this.state.notices}/>
          );
        else
            return <Login {...props} signIn = {this.signIn}/>
        }}/>
      <Route exact path="/" component={Login}/>
      <Route exact path="/notice/:id" render={(props)=>{
        return(<NoticePage {...props} notice={this.state.notices.find((notice)=>props.match.params.id === String(notice.id))}/>);
      }}/>
    </Switch>
    <Route path="/newNotice" component={NewNotice}/>  
    </div>
  );
  }
}

export default App;
