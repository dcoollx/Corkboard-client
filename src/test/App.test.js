import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer';
import Corkboard from '../components/main/announcements/corkboard';
import noticePage from '../components/main/announcements/NoticePage/noticePage';
import test from './demoData';
import NoticePage from "../components/main/announcements/NoticePage/noticePage";
import Landing from "../components/Landing/Landing";
//import ReactModal from "react-modal";

it("renders without crashing", () => {
  const div = document.createElement("div");
  div.id="root";
    ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe('major components',()=>{
  
  it('renders corkboard',() => {
  const div = document.createElement('div');
  ReactDOM.render(<Corkboard data={test.testNotices()}/>, div);
  ReactDOM.unmountComponentAtNode(div);
  });
})

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<BrowserRouter><App /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});