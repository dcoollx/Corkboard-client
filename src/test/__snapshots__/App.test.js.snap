// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<main
  className="App main-container "
>
  <nav
    className=" container col-full"
    id="header"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="logo"
          id="logo"
          src="logo.png"
        />
      </a>
      <h1
        className="home"
      >
        Not connected to an org
      </h1>
      <p
        style={
          Object {
            "fontSize": "40%",
            "margin": "0px",
          }
        }
      >
        Not connected to a team
      </p>
    </div>
    <div
      className="col-right"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        log in
      </a>
       | 
      <a
        href="/register"
        onClick={[Function]}
      >
        register
      </a>
    </div>
  </nav>
  <div
    className="container page"
    id="landing"
  >
    <div
      className="col-full small_container"
      id="landing_header"
    >
      <div
        className="col-full small_container"
      >
        <div
          className="col-1"
        >
          <h1>
            Corkboard
          </h1>
          <q>
            The Inter-Organization Message Management System
          </q>
        </div>
        <div
          className="landing_sub col-right no_mobile"
        >
          New here? 
          <a
            href="/register"
            onClick={[Function]}
          >
            Create An Account
          </a>
          <br />
          Come here often? 
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
    <div
      className="small_container col-full  "
    >
      <h3
        className="col-left"
      >
        Share messsages with your whole Organization or just a single team
      </h3>
      <img
        alt="connections"
        className="col-right"
        src="connections.png"
      />
    </div>
    <div
      className="small_container col-full  "
    >
      <img
        alt="pictures"
        className="col-left"
        src="picture.png"
      />
      <h3
        className="col-right"
      >
        Share photos and links
      </h3>
    </div>
    <div
      className="small_container col-full  "
    >
      <h3
        className="col-left"
      >
        Craft exciting notifications using HTML5 or markdown
      </h3>
      <img
        alt="web browser"
        className="col-right"
        src="browser.png"
      />
    </div>
    <footer
      className="small_container col-full "
    >
      <p
        className="col-1"
      >
        Demo Credentials: testuser, Password1!
      </p>
      <p
        className="col-right"
        id="copyright"
      >
         Â© 2019 
        <a
          href="https://dcoollx.github.io/Portfolio"
        >
          David Queen Jr
        </a>
      </p>
    </footer>
  </div>
</main>
`;
